### PR TITLE
Do not include dev website paths in search index of production website

### LINKED ISSUE #1680: Changelog / News Contents (pkgdown dev version)
Hi,

Level one headings are not rendered in the table of contents produced with the dev version of pkgdown.
With usual level 1 heading for release: https://umr1283.github.io/umr1283/news/index.html
image
With release version as level 2 headings: https://umr1283.github.io/umr1283/dev/news/index.html
image

Is that on purpose? if so, I am guessing this will break changelog page for many pkgdown website and will need to be reflected in {usethis} (e.g., use_version functions)

### PR DESCRIPTION
**Pull request by maelle**
Fix #1680

Thanks to report by @mcanouil in https://github.com/r-lib/pkgdown/issues/1680#issuecomment-838628384

Something to keep in mind too if one day pkgdown keeps website of different releases.

### CODE REVIEW

File: R/build-search-docs.R [hunk 1]
```diff
@@ -102,12 +102,18 @@ build_search <- function(pkg = ".",
...
+  paths <- fs::path_rel(
     fs::dir_ls(pkg$dst_path, glob = "*.html", recurse = TRUE),
     pkg$dst_path
   )
   paths <- paths[!paths %in% c("404.html", "articles/index.html", "reference/index.html")]
 
+  if (!pkg$development$in_dev) {
+    # do not include dev package website in search index
+    dev_destination <- meta_development(pkg$meta)$destination
+    paths <- paths[!grepl(paste0("^", dev_destination, "/"), paths)]
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/build-search-docs.R:114]: `paths[fs::path_dir(paths) != "dev"]` ?

And I think we can do this unconditionally.

[the remark this answers is not in the document]
Comment (maelle) [reply R/build-search-docs.R:114]: I'll remove the conditional but `path_dir()` won't work

``` r
fs::path_dir("dev/articles/bla.html")
#> [1] "dev/articles"
```

<sup>Created on 2021-05-12 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0.9000)</sup>

[the remark this answers is not in the document]
Comment (mcanouil) [reply R/build-search-docs.R]: Would fs::path_has_parent not be the answer to check if for a given path dev is a parent?

[the remark this answers is not in the document]
Comment (maelle) [reply R/build-search-docs.R]: thank you! I made this change in #1684 (and if that PR ends up being a bad idea I'll port the change here).

File: R/build-search-docs.R [hunk 2]
```diff
@@ -314,3 +308,13 @@ is_heading <- function(node) {
...
   paste0(toupper(substring(string, 1, 1)), substring(string, 2))
 }
+
+get_site_paths <- function(pkg) {
+  paths <- fs::path_rel(
+    fs::dir_ls(pkg$dst_path, glob = "*.html", recurse = TRUE),
+    pkg$dst_path
+  )
+  # do not include dev package website in search index / sitemap
+  dev_destination <- meta_development(pkg$meta)$destination
```
Comment (maelle) [inline R/build-search-docs.R:318]: using this as it might not be "dev"

Comment (hadley) [reply R/build-search-docs.R:318]: Is it always a relative path?

Comment (maelle) [reply R/build-search-docs.R:318]: yes if I understand 

https://github.com/r-lib/pkgdown/blob/23a04154b020bde5ed0cfee75d043ef5f604a649/R/package.r#L43

correctly

File: R/build-search-docs.R [hunk 3]
```diff
@@ -40,10 +40,7 @@ build_sitemap <- function(pkg = ".") {
 
   urls <- paste0(
```
Comment (hadley) [inline R/build-search-docs.R:41]: This can now fit on one line

File: R/build-search-docs.R [hunk 4]
```diff
@@ -314,3 +308,13 @@ is_heading <- function(node) {
 capitalise <- function(string) {
   paste0(toupper(substring(string, 1, 1)), substring(string, 2))
 }
+
+get_site_paths <- function(pkg) {
+  paths <- fs::path_rel(
```
Comment (hadley) [inline R/build-search-docs.R:313]: Can you please split this into multiple independent lines? e.g. `paths <- ...`, `paths_rel <- ...`

**Review by hadley — APPROVED** [review]

**Pull request merged**